torchtext/csrc/vectors.cpp

```diff
@@ -0,0 +1,68 @@
+#include <string>
+#include <torch/script.h>
+
+namespace torchtext {
+namespace {
+
+struct Vectors : torch::CustomClassHolder {
+private:
+  std::unordered_map<std::string, torch::Tensor> stovectors_;
+
+public:
+  // tokens_, vectors_, and unk_tensor_ holds the serialized params passed in
+  // during initialization. We need this because we need to be able to serialize
+  // the model so that we can save the scripted object. Pickle will get the
+  // serialized model from these members, thus they needs to be public.
+  std::vector<std::string> tokens_;
+  std::vector<torch::Tensor> vectors_;
+  torch::Tensor unk_tensor_;
+
+  explicit Vectors(std::vector<std::string> tokens,
+                   std::vector<torch::Tensor> vectors, torch::Tensor unk_tensor)
+      : tokens_(tokens), vectors_(vectors), unk_tensor_(unk_tensor) {
+    for (std::size_t i = 0; i < tokens.size(); i++) {
+      // guarding against size mismatch of vectors and tokens
+      if (i >= vectors.size()) {
+        break;
+      }
+
+      stovectors_[tokens[i]] = vectors_[i];
+    }
+  }
+
+  torch::Tensor GetItem(const std::string &token) const {
+    if (stovectors_.find(token) != stovectors_.end()) {
+      return stovectors_.at(token);
+    }
+    return unk_tensor_;
+  }
+};
+
+// Registers our custom class with torch.
+static auto vectors =
+    torch::class_<Vectors>("torchtext", "Vectors")
+        .def(torch::init<std::vector<std::string>, std::vector<torch::Tensor>,
+                         torch::Tensor>())
+        .def("GetItem", &Vectors::GetItem)
+        .def_pickle(
```

Comment:
`AddItem` func to attach custom symbols here?

Reply:
@zhangguanheng66  what should this do if we try to add an item for which the token already exists in our `vector`?

Reply:
That's a good question. One possibility is that users want to overwrite an existing token with a new vector. Since we make this func public, users should know what they are doing here. @cpuhrsch Do you think if we need a warning message here?

Reply:
okay I think thats fine then. the default behavior will be to overwrite the existing vector